Tools/Scripts/libraries/webkitscmpy/webkitscmpy/program/conflict.py

```diff
@@ -39,9 +41,40 @@ def parser(cls, parser, loggers=None):
         )
 
     @classmethod
-    def find_conflict_pr(cls, remote, branch):
-        for pr in remote.pull_requests.find(head=branch, opened=True):
-            return pr
+    def radar_for_id(cls, radar_id):
+        for tracker in Tracker._trackers:
+            if isinstance(tracker, radar.Tracker):
+                return tracker().client.radar_for_id(radar_id, additional_fields=['sourceChanges'])
+
+    @classmethod
+    def find_conflict_pr(cls, remote, radar_id):
+        """
+        Because we don't know what the target branch was just given the radar id,
+        we need to search for PRs with branches that start with the integration prefix
+        and have the first and last sha.
+        """
+        radar_obj = cls.radar_for_id(radar_id)
+        assert radar_obj, 'Could not fetch radar object for id {}'.format(radar_id)
+        shas = []
+
+        repo_name = remote.name if '/' not in remote.name else remote.name.split('/', 1)[-1]
+        for entry in radar_obj.sourceChanges.splitlines():
+            repo, action, sha = entry.split(', ')
+            if repo == repo_name:
+                shas.append(sha)
+
+        integration_branches = []
+        for prefix in ('ci', 'conflict'):
+            integration_branches.append("{}/{}/{}_{}".format(cls.INTEGRATION_BRANCH_PREFIX, prefix, shas[0][:10], shas[-1][:10]))
```

Comment:
We should be using 12 character shas, enshrined in the `Commit.HASH_LABEL_SIZE` variable (there's a whole explanation as to why, the TLDR is "birthday paradox, WebKit needs 12 character shas to guarantee uniqueness".

Reply:
ah that's right, let me update that here and in our automerge code

Reply:
https://stashweb.sd.apple.com/projects/SAFARI/repos/automerge/pull-requests/1725/overview